gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -336,9 +315,53 @@ private boolean hasValidationErrors() {
   }
 
   public Serializable getDraftResponses() {
-    HashMap<String, Response> hashMap = new HashMap<>();
-    hashMap.putAll(responses);
-    return hashMap;
+    return new HashMap<>(responses);
+  }
+
+  @Nullable
+  public String getFieldWaitingForPhoto() {
+    return fieldWaitingForPhoto;
+  }
+
+  public void setFieldWaitingForPhoto(@Nullable String fieldWaitingForPhoto) {
+    this.fieldWaitingForPhoto = fieldWaitingForPhoto;
+  }
+
+  public Observable<Pair<String, Optional<Bitmap>>> getLastPhotoResult() {
```

Comment:
Would it be worth creating a new class to encapsulate this result so that the fields could be accessed by name in OnPhotoResult() instead of .first and .second?

Reply:
Sure. Done!